taxonomy/utils.py

```diff
@@ -387,3 +387,28 @@ def apply_batching_to_translate_large_text(course_key, source_text):
     result['TranslatedText'] = translated_text
     result['SourceLanguageCode'] = source_language_code
     return result
+
+
+def get_whitelisted_course_skill_details(course_key):
+    """
+    Get whitelisted course skills with category and subcategory information.
+
+    Arguments:
+        course_key (str): Key of the course whose course skills need to be returned.
+
+    Returns:
+        (list<dict>): A list of all the course skills.
+    """
+    course_skills = CourseSkills.objects.filter(
+        course_key=course_key,
+        is_blacklisted=False,
+        skill__category__isnull=False,
```

Comment:
@DawoudSheraz - Do we need this check? I'm not sure but this way we will not get any skill which don't have category and sub category populated

Reply:
I am awaiting confirmation on this internally, but I suspect the missing category/sub-category will effect Algolia's search facet. https://2u-internal.atlassian.net/browse/PROD-2890?focusedCommentId=625188